www/js/app.js

```diff
@@ -1325,83 +1399,117 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
         var htmlCSS = htmlArticle.match(/<html[^>]*class\s*=\s*["']\s*([^"']+)/i);
         htmlCSS = htmlCSS ? htmlCSS[1] : '';
         
-        // Tell jQuery we're removing the iframe document: clears jQuery cache and prevents memory leaks [kiwix-js #361]
-        $('#articleContent').contents().remove();
-
         // Hide any alert box that was activated in uiUtil.displayFileDownloadAlert function
         $('#downloadAlert').hide();
 
-        var iframeArticleContent = document.getElementById('articleContent');
-        
-        iframeArticleContent.onload = function() {
-            iframeArticleContent.onload = function(){};
+        // Code below will run after we have written the new article to the articleContainer
+        var windowLoaded = function() {
             $("#articleList").empty();
             $('#articleListHeaderMessage').empty();
             $('#articleListWithHeader').hide();
             $("#prefix").val("");
-            
-            var iframeContentDocument = iframeArticleContent.contentDocument;
-            if (!iframeContentDocument && window.location.protocol === 'file:') {
-                alert("You seem to be opening kiwix-js with the file:// protocol, which is blocked by your browser for security reasons."
-                        + "\nThe easiest way to run it is to download and run it as a browser extension (from the vendor store)."
-                        + "\nElse you can open it through a web server : either through a local one (http://localhost/...) or through a remote one (but you need SSL : https://webserver/...)"
-                        + "\nAnother option is to force your browser to accept that (but you'll open a security breach) : on Chrome, you can start it with --allow-file-access-from-files command-line argument; on Firefox, you can set privacy.file_unique_origin to false in about:config");
-                return;
+            if (appstate.target === 'iframe' && !articleContainer.contentDocument && window.location.protocol === 'file:') {
+                if (!appstate.fileAccessWarning) {
```

Comment:
Because this warning about file:// access is longer now, with more user-friendly explanation, I have formatted it better and have used a simple DOM-based (Bootstrap-styled), non-blocking `systemAlert()` function. Arguably we should make all system alerts use this function.

Also, rather than bombing out at this point, I found that in Chrome and Firefox allow us to inject the HTML into a new window and show it fully styed. Chrome/Edge launches this window immediately and shows it, but Firefox blocks it by default as a popup. After the user allows the popup, they can try again and it will show just like in Chromium. I thought this was fun (and potentially useful)!